src/Core/Authorization/AuthorizationResolver.cs

```diff
@@ -438,14 +440,109 @@ public IEnumerable<string> GetAllowedExposedColumns(string entityName, string ro
     /// <summary>
-    /// Helper method to extract all claims available in the HttpContext's user object and add the claims
-    /// to the claimsInRequestContext dictionary to be used for claimType -> claim lookups.
+    /// Returns a dictionary (string, string) where a key is a claim's name and
+    /// a value is a claim's value.
+    /// Resolves scope/scp claims to a single string value with scopes delimited by spaces.
+    /// Resolves multiple instances of a claim into a JSON array mirroring the format
+    /// of the claim in the original JWT token. The JSON array's type depends on the value type
+    /// present in the original JWT token.
     /// </summary>
+    /// <remarks>
+    /// DotNet will resolve a claim with value type JSON array to multiple Claim objects with the same type and different values.
+    /// e.g. roles and groups claim, which are arrays of strings and are flattened to: roles: role1, roles: role2, groups: group1, groups: group2
+    /// Claims are name valued pairs and nothing more. https://github.com/dotnet/aspnetcore/issues/13647#issuecomment-527523224
+    /// The library that parses the jwt token into claims is the one that decides *how* to resolve the claims from the token's JSON payload.
+    /// DotNet flattens the claims into a list: https://github.com/dotnet/aspnetcore/blob/282bfc1b486ae235a3395150a8d53073a57b7f43/src/Security/Authentication/OAuth/src/JsonKeyClaimAction.cs#L39-L53
+    /// </remarks>
+    /// <param name="context">HttpContext which contains a ClaimsPrincipal</param>
+    /// <returns>Processed claims and claim values.</returns>
+    public static Dictionary<string, string> GetProcessedUserClaims(HttpContext? context)
+    {
+        Dictionary<string, string> processedClaims = new();
+
+        if (context is null)
+        {
+            return processedClaims;
+        }
+
+        Dictionary<string, List<Claim>> userClaims = GetAllAuthenticatedUserClaims(context);
+
+        foreach ((string claimName, List<Claim> claimValues) in userClaims)
+        {
+            // Special case to accommodate historical versions of Duende IdentityServer which emitted
+            // the scope claim as a JSON array in the access token.
+            // Entra ID emits the scope claim as 'scp' whose value is a string of values separated by spaces
+            // as described in the Entra ID Access Token Claims Reference
+            // When >1 scope/scp claim exists, DAB resolves the values into a single string where scopes are delimited by spaces.
+            // https://learn.microsoft.com/entra/identity-platform/access-token-claims-reference#payload-claims
+            // https://docs.duendesoftware.com/identityserver/v5/apis/aspnetcore/authorization/#scope-claim-format
+            if ((claimName == "scp" || claimName == "scope") && claimValues.Count > 1)
+            {
+                StringBuilder spaceDelimitedClaimPayload = new();
+                for (int i = 0; i < claimValues.Count - 1; i++)
```

Comment:
No limit and `Count` is type `int`.

Reply:
also removed this special handling of scp/scope so that DAB is no longer creating this space separated string. That space separated string is only preserved when using EntraID . so no need to recreated.

your question still applies to max count though. JWT tokens issue by identity providers are capped by max payload of HTTP header. If I were to put a cap on claim count, I'm uncertain what would be a "best" value without unintentionally truncating data a developer expects to be present.  For example, the `groups` claim is capped at Entra ID at 250 groups, before the groups claim values are replaced with a "Distributed claim" which means the value is a URL to microsoft graph where the app can query for a list of groups because the number was too large to include in a token.